doc/panama_ffi.md

```diff
@@ -195,30 +195,34 @@ Note that, since the function `strlen` is part of the standard C library, which
 One we have obtained the downcall native method handle, we can just use it as any other method handle:
 
 ```java
-long len = strlen.invokeExact(CLinker.toCString("Hello").address()); // 5
+try (ResourceScope scope = ResourceScope.ofConfined()) {
+    long len = strlen.invokeExact(CLinker.toCString("Hello", scope).address()); // 5
+}
 ```
 
-Here we are using one of the helper methods in `CLinker` to convert a Java string into an off-heap memory segment which contains a `NULL` terminated C string. We then pass that segment to the method handle and retrieve our result in a Java `long`. Note how all this has been possible *without* any piece of intervening native code — all the interop code can be expressed in (low level) Java.
+Here we are using one of the helper methods in `CLinker` to convert a Java string into an off-heap memory segment which contains a `NULL` terminated C string. We then pass that segment to the method handle and retrieve our result in a Java `long`. Note how all this has been possible *without* any piece of intervening native code — all the interop code can be expressed in (low level) Java. Note also how we used an explicit resource scope to control the lifecycle of the allocated C string; while using the implicit *default scope* is an option, extra care must be taken when using segments featuring implicitly deallocation which are then converted into `MemoryAddress` instances: since the address is eventually converted (by the linker support) into a raw Java long, there is no guarantee that the memory segment would be kept *reachable* for the entire duration of the native call.
 
 The `CLinker` interfaces also supports linking of native function without an address known at link time; when that happens, an address must be provided when the method handle returned by the linker is invoked - this is very useful to support *virtual calls*. For instance, the above code can be rewritten as follows:
 
 ```java
-MethodHandle strlen_virtual = CLinker.getInstance().downcallHandle(
+MethodHandle strlen_virtual = CLinker.getInstance().downcallHandle( // address parameter missing!
 		MethodType.methodType(long.class, MemoryAddress.class),
         FunctionDescriptor.of(C_LONG, C_POINTER)
 );
 
-long len = strlen_virtual.invokeExact(
-	LibraryLookup.ofDefault().lookup("strlen").get()
-	CLinker.toCString("Hello").address()
-); // 5
+try (ResourceScope scope = ResourceScope.ofConfined()) {
+    long len = strlen_virtual.invokeExact(
+        LibraryLookup.ofDefault().lookup("strlen").get() // address provided here!
+        CLinker.toCString("Hello").address()
```

Comment:
The scope is not being used for the string here.

Reply:
Looking at the full source code at the end, this also seems to mismatch with that.